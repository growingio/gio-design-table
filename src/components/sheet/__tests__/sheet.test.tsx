import { render } from '@testing-library/react';
import { S2Options } from '@antv/s2';
import * as mockDataConfig from './data/simple-data.json';
import { DataTable } from '..';

const s2Options: S2Options = {
  width: 600,
  height: 600,
  hierarchyType: 'grid',
};
describe('Spread Sheet Tests', () => {
  const hasScrollBar = (container: HTMLElement) => {
    const s2Container = container.querySelector('.gio-d-table-container') as HTMLElement;
    const res = (
      (s2Container.scrollWidth > s2Container.clientWidth ||
        document.body.scrollWidth > window.innerWidth) &&
      window.getComputedStyle(s2Container).overflow !== 'hidden'
    );
    return res;
  };

  describe('Mount Sheet tests', () => {

    test('should display scroll bar if s2Options.width more than browser window width', () => {
      const { container } = render(<div><DataTable
        options={{
          ...s2Options,
          width: window.innerWidth + 100,
        }}
        adaptive={false}
        dataConfig={mockDataConfig as any}
      /></div>);
      expect(container).toMatchSnapshot();
      expect(hasScrollBar(container)).toBeFalsy();
    });

  });
});