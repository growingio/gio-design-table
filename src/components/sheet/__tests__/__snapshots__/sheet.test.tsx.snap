// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spread Sheet Tests Mount Sheet tests should display scroll bar if s2Options.width more than browser window width 1`] = `
<div>
  <div>
    <div
      class="gio-loading-wrapper-loading"
    >
      <div
        class="gio-loading-container gio-loading-container-blur-white"
      >
        <div
          class="gio-d-table-wrapper"
        >
          <div
            class="gio-d-table-header gio-d-table-header"
            style="width: 1124px;"
          >
            <div
              class="gio-d-table-header-heading"
            >
              <div
                class="gio-d-table-header-heading-left"
              />
              <span
                class="gio-d-table-header-heading-extra"
              />
            </div>
          </div>
          <div
            class="gio-d-table-container"
          >
            <canvas
              height="600"
              style="width: 1124px; height: 600px; display: block;"
              width="1124"
            />
          </div>
        </div>
      </div>
    </div>
     
  </div>
</div>
`;
